packages/plugin/vite/src/VitePlugin.ts

```diff
@@ -0,0 +1,160 @@
+import fs from 'node:fs';
+import http from 'node:http';
+import path from 'node:path';
+
+import { namedHookWithTaskFn, PluginBase } from '@electron-forge/plugin-base';
+import { ForgeMultiHookMap, StartResult } from '@electron-forge/shared-types';
+import debug from 'debug';
+// eslint-disable-next-line node/no-extraneous-import
+import { RollupWatcher } from 'rollup';
+import { default as vite } from 'vite';
+
+import { VitePluginConfig } from './Config';
+import ViteConfigGenerator from './ViteConfig';
+
+const d = debug('electron-forge:plugin:vite');
+
+export default class VitePlugin extends PluginBase<VitePluginConfig> {
+  name = 'vite';
+
+  private isProd = false;
+
+  // The root of the Electron app
+  private projectDir!: string;
+
+  // Where the Vite output is generated. Usually `${projectDir}/.vite`
+  private baseDir!: string;
+
+  private _configGenerator!: ViteConfigGenerator;
+
+  private watchers: RollupWatcher[] = [];
+
+  private servers: http.Server[] = [];
+
+  init = (dir: string): void => {
+    this.setDirectories(dir);
+
+    d('hooking process events');
+    process.on('exit', (_code) => this.exitHandler({ cleanup: true }));
+    process.on('SIGINT' as NodeJS.Signals, (_signal) => this.exitHandler({ exit: true }));
+  };
+
+  setDirectories = (dir: string): void => {
+    this.projectDir = dir;
+    this.baseDir = path.join(dir, '.vite');
+  };
+
+  get configGenerator(): ViteConfigGenerator {
+    return (this._configGenerator ??= new ViteConfigGenerator(this.config, this.projectDir, this.isProd));
+  }
+
+  getHooks = (): ForgeMultiHookMap => {
+    return {
+      prePackage: [
+        namedHookWithTaskFn<'prePackage'>(async () => {
+          this.isProd = true;
+          fs.rmSync(this.baseDir, { recursive: true, force: true });
+
+          await this.build();
+          await this.buildRenderer();
```

Comment:
Is concurrency possible here?
```suggestion
          await Promise.all([this.build(), this.buildRenderer()]);
```